odoo/addons/base/wizard/base_partner_merge.py

```diff
@@ -147,50 +146,52 @@ def _update_foreign_keys(self, src_partners, dst_partner):
                                 "%(column)s" = %%s AND
                                 ___tu.%(value)s = ___tw.%(value)s
                         )""" % query_dic
-                for partner in src_partners:
-                    self._cr.execute(query, (dst_partner.id, partner.id, dst_partner.id))
+                for record in src_records:
+                    self._cr.execute(query, (dst_record.id, record.id, dst_record.id))
             else:
                 try:
                     with mute_logger('odoo.sql_db'), self._cr.savepoint():
                         query = 'UPDATE "%(table)s" SET "%(column)s" = %%s WHERE "%(column)s" IN %%s' % query_dic
-                        self._cr.execute(query, (dst_partner.id, tuple(src_partners.ids),))
+                        self._cr.execute(query, (dst_record.id, tuple(src_records.ids)))
                 except psycopg2.Error:
                     # updating fails, most likely due to a violated unique constraint
                     # keeping record with nonexistent partner_id is useless, better delete it
                     query = 'DELETE FROM "%(table)s" WHERE "%(column)s" IN %%s' % query_dic
-                    self._cr.execute(query, (tuple(src_partners.ids),))
+                    self._cr.execute(query, (tuple(src_records.ids),))
 
     @api.model
-    def _update_reference_fields(self, src_partners, dst_partner):
-        """ Update all reference fields from the src_partner to dst_partner.
-            :param src_partners : merge source res.partner recordset (does not include destination one)
-            :param dst_partner : record of destination res.partner
+    def _update_reference_fields_generic(self, referenced_model, src_records, dst_record, additional_update_records=[]):
```

Comment:
```suggestion
    def _update_reference_fields_generic(self, referenced_model, src_records, dst_record, additional_update_records=None):
        additional_update_records = additional_update_records or []
```
Mutable default arguments are not recommended (see [here](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments))

Reply:
Done. Thanks again.